pkg/core/proxy/integrations/mongo/operation.go

```diff
@@ -735,8 +858,9 @@ func (r *opReply) TransactionDetails() *TransactionDetails {
 	return nil
 }
 
-func encodeOpReply(reply *models.MongoOpReply, logger *zap.Logger) (*opReply, error) {
+func encodeOpReply(mongoRequests models.MongoRequest, expected models.MongoRequest, reply *models.MongoOpReply, logger *zap.Logger) (*opReply, error) {
```

Comment:
may be change the parameter names to current request, recorded request, recorded response something like that, it gives more clarity.